src/classic/clvm_tools/cmds.rs

```diff
@@ -244,20 +246,170 @@ pub fn brun(args: &[String]) {
+                let mut function_entry = BTreeMap::new();
+                function_entry.insert(
+                    "Function-Name".to_string(),
+                    YamlElement::String(runner.running[run_idx].function_name.clone()),
+                );
+                let mut arg_values = BTreeMap::new();
+                for (k, v) in runner.running[run_idx].named_args.iter() {
+                    arg_values.insert(k.clone(), YamlElement::String(format!("{v}")));
+                }
+                function_entry.insert(
+                    "Function-Args".to_string(),
+                    YamlElement::Subtree(arg_values),
+                );
+                let mut info_values = BTreeMap::new();
+                for (k, v) in info.iter() {
+                    info_values.insert(k.clone(), YamlElement::String(v.clone()));
+                }
+                function_entry.insert("Output".to_string(), YamlElement::Subtree(info_values));
+
+                // Put in this entry on the current frame.
+                let os_last = output_stack.len() - 1;
+                output_stack[os_last].push(function_entry);
+
+                // If we're showing too many frames, ensure that children
+                // are in their parent entries.
+                while running_frames < output_stack.len() {
+                    let take_stack = output_stack
+                        .pop()
+                        .unwrap()
+                        .iter()
+                        .map(|e| YamlElement::Subtree(e.clone()))
+                        .collect();
+                    let mut inner_run_item: BTreeMap<String, YamlElement> = BTreeMap::new();
+                    inner_run_item.insert("Compute".to_string(), YamlElement::Array(take_stack));
+                    let os_last = output_stack.len() - 1;
+                    output_stack[os_last].push(inner_run_item);
+                }
+            }
+            Ok(HierarchialStepResult::Info(None)) => {
+                // Nothing
+            }
+            Ok(HierarchialStepResult::Done(Some(info))) => {
+                let mut done_output = BTreeMap::new();
+                for (k, v) in info.iter() {
+                    done_output.insert(k.clone(), YamlElement::String(v.clone()));
+                }
+                let os_last = output_stack.len() - 1;
+                output_stack[os_last].push(done_output);
+            }
+            Ok(HierarchialStepResult::Done(None)) => {
+                // Nothing
+            }
+            Err(RunFailure::RunErr(l, e)) => {
+                println!("Runtime Error: {l}: {e}");
+                break;
+            }
+            Err(RunFailure::RunExn(l, e)) => {
+                println!("Raised exception: {l}: {e}");
+                break;
+            }
+        }
+    }
+
+    // Move out of output_stack nicely.
+    let mut result = Vec::new();
+    swap(&mut result, &mut output_stack[0]);
+    result
```

Comment:
```suggestion
    output_stack[0]
```

Reply:
output_stack is used here so we can't move directly out of it.  This swap avoids a clone.